debian/patches/backport-ubuntu-pro-rename.patch

```diff
@@ -0,0 +1,4187 @@
++    except Exception:
++        util.logexc(LOG, "Failed to install ubuntu-advantage-tools")
++        raise
++
++
++def _should_auto_attach(pro_section: dict) -> bool:
++    disable_auto_attach = bool(
++        pro_section.get("features", {}).get("disable_auto_attach", False)
++    )
++    if disable_auto_attach:
++        return False
++
++    # pylint: disable=import-error
++    from uaclient.api.exceptions import UserFacingError
++    from uaclient.api.u.pro.attach.auto.should_auto_attach.v1 import (
++        should_auto_attach,
++    )
++
++    # pylint: enable=import-error
++
++    try:
++        result = util.log_time(
++            logfunc=LOG.debug,
++            msg="Checking if the instance can be attached to Ubuntu Pro",
++            func=should_auto_attach,
++        )
++    except UserFacingError as ex:
++        LOG.debug("Error during `should_auto_attach`: %s", ex)
++        LOG.warning(ERROR_MSG_SHOULD_AUTO_ATTACH)
++        return False
++    return result.should_auto_attach
++
++
++def _attach(pro_section: dict):
++    token = pro_section.get("token")
++    if not token:
++        msg = "`ubuntu_pro.token` required in non-Pro Ubuntu instances."
++        LOG.error(msg)
++        raise RuntimeError(msg)
++    enable_beta = pro_section.get("enable_beta")
++    if enable_beta:
++        LOG.debug(
++            "Ignoring `ubuntu_pro.enable_beta` services in Pro attach: %s",
++            ", ".join(enable_beta),
++        )
++    configure_pro(token=token, enable=pro_section.get("enable"))
++
++
++def _auto_attach(pro_section: dict):
++
++    # pylint: disable=import-error
++    from uaclient.api.exceptions import AlreadyAttachedError, UserFacingError
++    from uaclient.api.u.pro.attach.auto.full_auto_attach.v1 import (
++        FullAutoAttachOptions,
++        full_auto_attach,
++    )
++
++    # pylint: enable=import-error
++
++    enable = pro_section.get("enable")
++    enable_beta = pro_section.get("enable_beta")
++    options = FullAutoAttachOptions(
++        enable=enable,
++        enable_beta=enable_beta,
++    )
++    try:
++        util.log_time(
++            logfunc=LOG.debug,
++            msg="Attaching to Ubuntu Pro",
++            func=full_auto_attach,
++            kwargs={"options": options},
++        )
++    except AlreadyAttachedError:
++        if enable_beta is not None or enable is not None:
++            # Only warn if the user defined some service to enable/disable.
++            LOG.warning(
++                "The instance is already attached to Pro. Leaving enabled"
++                " services untouched. Ignoring config directives"
++                " ubuntu_pro: enable and enable_beta"
++            )
++    except UserFacingError as ex:
++        msg = f"Error during `full_auto_attach`: {ex.msg}"
++        LOG.error(msg)
++        raise RuntimeError(msg) from ex
++
++
++def handle(
++    name: str, cfg: Config, cloud: Cloud, log: logging.Logger, args: list
```

Comment:
Backport had to add back the `log` param to handle() signature to align w/ bionic